src/components/DownloadZIP.tsx

```
@@ -0,0 +1,366 @@
+  const [progress, setProgress] = useState(0);
+  const [statusText, setStatusText] = useState("");
+  const toast = useToast();
+
+  // Helper function to sanitize file names
+  const sanitizeFileName = (name: string | number) =>
+    String(name)
+      .replace(/[\\/:*?"<>|]/g, "_")
+      .replace(/\s+/g, "_");
+
+  // Helper function to convert application data to CSV
+  const convertApplicationToCSV = (applicationData: any): string => {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)